pkg/identity/api/v1/service.go

```diff
@@ -41,14 +41,45 @@ func (s *Service) Close() {
 	s.log.Info("closed")
 }
 
+/*
+Algorithm:
+
+Note - the inbox_log table has global ordering via a serial sequence_ID, and an index by inbox_ID
```

Comment:
I think we might be able to get to the same outcome more simply if we just lean on Postgres' serializable transactions. Would look something like this pseudo-sql

```sql
BEGIN
SELECT * FROM inbox_log FOR UPDATE WHERE inbox_id = ?;
-- Take the selected rows and send them to the MLS validation service to validate and get the results
-- If invalid, abort the transaction and return an error

-- For each updated address, update the address_lookup_cache and address_log
INSERT INTO address_log VALUES(...)
UPDATE address_lookup_cache SET inbox_id = ? WHERE address = ?
COMMIT
```

If another update is attempting to write to the address lookup cache, you should hit a serialization error and be forced to retry.

Reply:
You can read more about the guarantees of serializable transactions [here](https://www.thenile.dev/blog/transaction-isolation-postgres)

Reply:
My hang-up with transactions was that I was uncertain if the use of a global SERIAL ID would cause all transactions to block on each other whenever a row was inserted. From our discussion, we found that sequences increment outside of transactions, and if a transaction rolls back, the sequence can have gaps - which is exactly the type of behavior we want.

Reply:
Coming back to this days later, this causes a different problem - the sequence number is not guaranteed to increase in the same 'serialization order' as the transactions: https://stackoverflow.com/questions/52432459/postgresql-serialized-inserts-interleaving-sequence-numbers

Thinking through how we can fix this and will put up another PR